kernel/src/driver/virtio/sysfs.rs

```diff
@@ -165,19 +190,10 @@ impl VirtIODeviceManager {
     pub fn device_add(&self, dev: Arc<dyn VirtIODevice>) -> Result<(), SystemError> {
         dev.set_bus(Some(Arc::downgrade(&(virtio_bus() as Arc<dyn Bus>))));
         device_manager().device_default_initialize(&(dev.clone() as Arc<dyn Device>));
-        let drv = dev.driver().ok_or(SystemError::EINVAL)?;
 
-        let virtio_drv = drv.cast::<dyn VirtIODriver>().map_err(|_| {
-            error!(
-                "VirtIODeviceManager::device_add() failed: device.driver() is not a VirtioDriver. Device: '{:?}'",
-                dev.name()
-            );
-            SystemError::EINVAL
-        })?;
         let virtio_index = VIRTIO_DEVICE_INDEX_MANAGER.alloc();
         dev.set_virtio_device_index(virtio_index);
         dev.set_device_name(format!("virtio{}", virtio_index.data()));
-        virtio_drv.probe(&dev)?;
 
         device_manager().add_device(dev.clone() as Arc<dyn Device>)?;
```

Comment:
比如这种地方，在较新的RA版本中会报错，使用intertrait cast后可消除该报错：
```suggestion
        device_manager().add_device(dev.cast::<dyn Device>().unwrap())?;
```